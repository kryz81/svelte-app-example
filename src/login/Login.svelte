<script>
  // imports
  import { login } from "../auth/auth";

  // state
  let username = "";
  let password = "";
  let submitted = false;
  const errMsg = "Invalid username/password";

  // actions
  const submit = () => {
    submitted = true;
    if (err) {
      return;
    }
    login(username);
  };

  const reset = () => (submitted = false);

  // computed
  $: err = submitted && (username !== "Kryz" || password !== "12345");
</script>

<div class="row">
  <div class="col-md-4 offset-md-4">
    {#if err}
      <div class="alert alert-danger" role="alert">{errMsg}</div>
    {/if}
    <form on:submit|preventDefault={submit}>
      <div class="form-group">
        <label>Username</label>
        <input
          bind:value={username}
          on:input={reset}
          type="text"
          class="form-control"
          data-testid="username" />
      </div>
      <div class="form-group">
        <label>Password</label>
        <input
          bind:value={password}
          on:input={reset}
          type="password"
          class="form-control"
          data-testid="password" />
      </div>
      <div class="text-center">
        <button type="submit" class="btn btn-primary">Login</button>
      </div>
    </form>
  </div>
</div>
