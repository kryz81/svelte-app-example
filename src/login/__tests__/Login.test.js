import {
  render,
  cleanup,
  fireEvent,
  waitForElement
} from "@testing-library/svelte";

import Login from "../Login.svelte";

window.location.reload = jest.fn();

beforeEach(() => {
  cleanup();
});

describe("Login", () => {
  it("renders component", () => {
    const { container } = render(Login);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("does not login and shows error message on invalid credentials", async () => {
    // given
    const { getByText, container } = render(Login);

    // when
    fireEvent.click(getByText("Login"));

    const errMessage = await waitForElement(
      () => container.querySelector("div.alert").innerHTML
    );

    // then
    expect(errMessage).toBe("Invalid username/password");
  });
});
