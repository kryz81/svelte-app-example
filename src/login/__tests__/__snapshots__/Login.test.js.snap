// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders component 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-md-4 offset-md-4"
    >
       
      <form>
        <div
          class="form-group"
        >
          <label>
            Username
          </label>
           
          <input
            class="form-control"
            data-testid="username"
            type="text"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label>
            Password
          </label>
           
          <input
            class="form-control"
            data-testid="password"
            type="password"
          />
        </div>
         
        <div
          class="text-center"
        >
          <button
            class="btn btn-primary"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
